packages/sdk/src/utils/gas.ts

```diff
@@ -0,0 +1,182 @@
+
+// This function is here to accomodate instances where a network has a minimum BaseBlockFee
+export function adjustBaseBlockFee(chainId: number, baseBlockFee: bigint): bigint {
+  const network: string = getNetworkByChainId(chainId).key
+  // Avalanche has a minimum BaseBlockFee of 25 GWEI
+  // https://docs.avax.network/quickstart/transaction-fees#base-fee
+  if (
+    (network === networks.avalanche.key || network === networks.avalancheTestnet.key) &&
+    baseBlockFee < BigInt(25000000000)
+  ) {
+    return BigInt(25000000000)
+  }
+
+  return baseBlockFee
+}
+
+// Implemented from https://eips.ethereum.org/EIPS/eip-1559
+export function calculateNextBlockFee(parent: Block): bigint {
+  const zero: bigint = BigInt(0)
+  if (parent.baseFeePerGas === undefined) {
+    return zero
+  }
+
+  const one: bigint = BigInt(1)
+  const elasticityMultiplier: bigint = BigInt(2)
+  const baseFeeMaxChangeDenominator: bigint = BigInt(8)
+  const baseFeePerGas: bigint = parent.baseFeePerGas!
+  const parentGasTarget: bigint = parent.gasLimit / elasticityMultiplier
+  if (parent.gasUsed === parentGasTarget) {
+    return baseFeePerGas
+  }
+
+  let gasUsedDelta: bigint
+  let baseFeeDelta: bigint
+  if (parent.gasUsed > parentGasTarget) {
+    // If the parent block used more gas than its target, the baseFee should increase.
+    gasUsedDelta = parent.gasUsed - parentGasTarget
+    baseFeeDelta = (baseFeePerGas * gasUsedDelta) / parentGasTarget / baseFeeMaxChangeDenominator
+    if (one > baseFeeDelta) {
+      baseFeeDelta = one
+    }
+
+    return baseFeePerGas + baseFeeDelta
+  }
+
+  // Otherwise if the parent block used less gas than its target, the baseFee should decrease.
+  gasUsedDelta = parentGasTarget - parent.gasUsed
+  baseFeeDelta = (baseFeePerGas * gasUsedDelta) / parentGasTarget / baseFeeMaxChangeDenominator
+  return baseFeePerGas - baseFeeDelta
+}
+
+export function updateGasPricing(chainId: number, block: Block, gasPricing: GasPricing): GasPricing {
+  if (block.baseFeePerGas) {
+    gasPricing.isEip1559 = true
+    gasPricing.nextBlockFee = adjustBaseBlockFee(chainId, calculateNextBlockFee(block))
+    gasPricing.maxFeePerGas = gasPricing.nextBlockFee
+
+    if (gasPricing.nextPriorityFee === null) {
+      gasPricing.gasPrice = gasPricing.nextBlockFee
+    } else {
+      gasPricing.maxFeePerGas = gasPricing.nextBlockFee! + gasPricing.nextPriorityFee!
+      gasPricing.gasPrice = gasPricing.maxFeePerGas
+    }
+
+    if (block.transactions.length > 0) {
+      if (typeof block.transactions[0] !== 'string') {
+        // we have a Block with transactions
+        for (let i = 0, l = block.transactions.length; i < l; i++) {
+          const tx = block.transactions[i] as Transaction
+          if (gasPricing.isEip1559) {
+            // set current tx priority fee
+            let priorityFee: bigint = BigInt(0)
+            let remainder: bigint
+
+            if (tx.maxFeePerGas === undefined || tx.maxPriorityFeePerGas === undefined) {
```

Comment:
We can certainly improve this logic, there are a bunch of ifs/elses, but I guess that could be with another PR